### ISSUE TITLE
Compliance checks do not cast values as needed

### ISSUE THREAD

**Issue by nllong** [SEED-platform/seed#3788]
**Describe the bug**

Compliance reports don't load

**Expected Behavior**

Should be able to visualize the compliance page

**Actual Behavior**

```
Traceback (most recent call last):
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/sentry_sdk/integrations/django/views.py", line 67, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/nlong/.pyenv/versions/3.9.7-seed/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/nlong/working/seed/seed/seed/decorators.py", line 212, in _wrapped
    return fn(self, request, *args, **kwargs)
  File "/Users/nlong/working/seed/seed/seed/utils/api.py", line 149, in _wrapped
    return fn(self, request, *args, **kwargs)
  File "/Users/nlong/working/seed/seed/seed/decorators.py", line 148, in wrapper
    response = func(self, request, *args, **kwargs)
  File "/Users/nlong/working/seed/seed/seed/lib/superperms/orgs/decorators.py", line 198, in _wrapped
    return fn(self, request, *args, **kwargs)
  File "/Users/nlong/working/seed/seed/seed/views/v3/compliance_metrics.py", line 242, in evaluate
    response = compliance_metric.evaluate()
  File "/Users/nlong/working/seed/seed/seed/models/compliance_metrics.py", line 140, in evaluate
    properties[p['property_view_id']] = self._calculate_compliance(p, metric['energy_bool'], 'energy')
  File "/Users/nlong/working/seed/seed/seed/models/compliance_metrics.py", line 196, in _calculate_compliance
    differential = target_val - actual_val
TypeError: unsupported operand type(s) for -: 'str' and 'float'
```

**Steps to Reproduce**

* import GHG or EUI target data as a string. There is no really easy way to enforce this, but it happened when using the `py-seed` library. 
* Set the column data type to GHG or EUI or float
* Set up the compliance metric to use the target field.
* Try to load the Insights page
* Page doesn't load with little helpful data, console log is above

**Instance Information**

**Additional Context**

**Comment by RDmitchell** [comment]
@nllong -- can this issue be closed?

**Comment by nllong** [comment]
yes, this has been fixed per #3789
